hudi-sync/hudi-sync-common/src/main/java/org/apache/hudi/sync/common/HoodieSyncConfig.java

```diff
@@ -94,28 +94,22 @@ public class HoodieSyncConfig extends HoodieConfig {
 
   public static final ConfigProperty<String> META_SYNC_PARTITION_EXTRACTOR_CLASS = ConfigProperty
       .key("hoodie.datasource.hive_sync.partition_extractor_class")
-      .defaultValue("org.apache.hudi.hive.MultiPartKeysValueExtractor")
+      .defaultValue("org.apache.hudi.hive.NonPartitionedExtractor")
       .withInferFunction(cfg -> {
-        Option<String> partitionFieldsOpt = Option.ofNullable(cfg.getString(HoodieTableConfig.PARTITION_FIELDS))
-            .or(() -> Option.ofNullable(cfg.getString(KeyGeneratorOptions.PARTITIONPATH_FIELD_NAME)));
-        if (!partitionFieldsOpt.isPresent()) {
+        Option<String> partitionFieldsOpt = Option.ofNullable(cfg.getString(META_SYNC_PARTITION_FIELDS));
```

Comment:
META_SYNC_PARTITION_FIELDS is inferred from HoodieTableConfig.PARTITION_FIELDS and KeyGeneratorOptions.PARTITIONPATH_FIELD_NAME, so I think it is ok to infrer META_SYNC_PARTITION_EXTRACTOR_CLASS from META_SYNC_PARTITION_FIELDS.

Reply:
the problem is when instantiating the hoodie sync config we cannot guarantee META_SYNC_PARTITION_FIELDS is inferred before META_SYNC_PARTITION_EXTRACTOR_CLASS. so we want to still check the table and key gen configs.

Reply:
i think it does make sense to infer from META_SYNC_PARTITION_FIELDS as users can manual set it, which should align with the partition extractor. so can you make it the 1st, then fallback to table, and then key gen

Reply:
Ok.